arangod/Replication2/ReplicatedState/LeaderStateManager.tpp

```diff
@@ -191,7 +191,8 @@ auto LeaderStateManager<S>::recoverEntries() noexcept
         factory->constructLeader(std::move(core));
 
     data.state = machine;
-
+    TRI_ASSERT(data.state != nullptr);
```

Comment:
I think it's good enough. Of course, a Production Assert wouldn't hurt, but `data.state` is the leader, as constructed by the factory just above. Since it is instantiated right before that, I don't see how it can be gone. However, if someone creates a factory which can return nullptr, then we have a problem, but for that a TRI_ASSERT looks good enough.